### PR TITLE
New traitor item: Point Defence Rifle

### PR DESCRIPTION
**Pull request by DrSirKnight**
## About The Pull Request

Adds a new munitions tech exclusive syndicate gun. It costs 12 TC and uses PDC rounds for ammo.
Does slightly less damage than the bolt action rifle, because of how common PDC rounds are. (50 instead of 60)

## Why It's Good For The Game

Gives one of the most popular NSV exclusive jobs a traitor item.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://user-images.githubusercontent.com/54851612/203846671-073e1b21-4373-453e-9c6a-5ff7bd54a8a7.mp4

</details>

## Changelog
:cl:
add: Point Defence Rifle added to the traitor uplink
imageadd: Added sprites for the Point Defence Rifle
/:cl:

### CODE REVIEW

File: nsv13/code/modules/munitions/ammunition/magazines.dm [hunk 1]
```diff
@@ -55,12 +55,20 @@
 	caliber = "mm30.12"
 	max_ammo = 300
 
+/obj/item/ammo_box/magazine/nsv/pdc/rifle
+	max_ammo = 1
+
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/munitions/ammunition/magazines.dm:60]:
```suggestion
/obj/item/ammo_box/magazine/internal/boltaction/pdc
	max_ammo = 3
```
You can probably give them more bullets in the internal magazine, and it should also be a subtype of /magazine/internal as it's inside the gun. (this also lets them load in multiple bullets at once, which _can_ be changed)

File: nsv13/code/modules/uplink/uplink_items.dm [hunk 2]
```diff
@@ -6,3 +6,9 @@
 			much to overload the reactor with these..."
 	item = /obj/item/fuel_rod/material/telecrystal
 	cost = 7
+/datum/uplink_item/role_restricted/pdc_rifle
+	name = "Point Defence Rifle"
```
Comment (Bokkiewokkie) [inline nsv13/code/modules/uplink/uplink_items.dm:10]:
```suggestion

/datum/uplink_item/role_restricted/pdc_rifle
	name = "Point Defence Rifle"
```
Leaving some space between different things helps with code readability

File: nsv13/code/game/objects/items/custom_guns.dm [hunk 3]
```diff
@@ -139,3 +139,21 @@
 	ammo_type = /obj/item/ammo_casing/shotgun/buckshot
 	max_ammo = 6
 	multiple_sprites = 1
+
+/obj/item/gun/ballistic/rifle/boltaction/pdc
+	name = "\improper Point Defence Rifle"
+	desc = "A rifle made to fire PDC rounds."
+	icon = 'nsv13/icons/obj/guns/projectile.dmi'
+	icon_state = "pdcrifle"
+	item_state = "pdcrifle"
```
Comment (Bokkiewokkie) [inline nsv13/code/game/objects/items/custom_guns.dm:148]: If you're not going to add a custom back sprite I recommend using one of a different gun, so it's at least recognizable on the character when equipped. (I'd say that just using the mosin one is fine, as it is similar in functionality and damage)
To just use the default /rifle/boltaction back sprite you can remove the item_state:
```suggestion
```

File: nsv13/code/game/objects/items/custom_guns.dm [hunk 4]
```diff
@@ -139,3 +139,21 @@
...
+	lefthand_file = 'nsv13/icons/mob/inhands/weapons/guns_lefthand.dmi'
+	righthand_file = 'nsv13/icons/mob/inhands/weapons/guns_righthand.dmi'
+	slot_flags = ITEM_SLOT_BACK
+	mag_type = /obj/item/ammo_box/magazine/nsv/pdc/rifle
+	can_bayonet = TRUE
+	knife_x_offset = 27
+	knife_y_offset = 13
+	w_class = 3 // It's a syndicate weapon so weight is reduced to allow it to be hidden in backpacks
```
Comment (Bokkiewokkie) [inline nsv13/code/game/objects/items/custom_guns.dm:156]: Debatable if this should be the case, it's a pretty powerful weapon.

**Review by Bokkiewokkie — CHANGES_REQUESTED** [review]
Good first feature PR 👍 
Let me know if you need any help regarding the comments I left.

File: nsv13/code/game/objects/items/custom_guns.dm (hunk 4 above)
Comment (Karmic-Skink) [reply nsv13/code/game/objects/items/custom_guns.dm:156]: Please also use actual defines for the weight class rather than just numbers.

**Comment by Oscari-pog** [comment]
Now add gauss rifle.

File: nsv13/code/game/objects/items/custom_guns.dm (hunk 4 above)
Comment (DrSirKnight) [reply nsv13/code/game/objects/items/custom_guns.dm:156]: If the weight class was higher then the rifle would be a downgrade to the revolver in almost every scenario, considering the revolver does slightly more damage, has ammo that can be printed, has a 6(?) round magazine and can be stored in the backpack. That's the only reason I lowered the weight class, but if you still think I should change it then I'll go ahead and do that.

Comment (Bokkiewokkie) [reply nsv13/code/game/objects/items/custom_guns.dm:156]: > If the weight class was higher then the rifle would be a downgrade to the revolver in almost every scenario, considering the revolver does slightly more damage, has ammo that can be printed, has a 6(?) round magazine and can be stored in the backpack. That's the only reason I lowered the weight class, but if you still think I should change it then I'll go ahead and do that.

It should still use a define like Karmic said either way, but that makes sense then.
Please look at my other comments as well.

**Review by Bokkiewokkie — APPROVED** [review]

**Pull request merged**